### ISSUE TITLE
SA module 

### ISSUE THREAD

**Issue by hfutzzw** [HRanWang/Spatial-Attention#3]
I notice that spatial attention module implemented in this repository is different with paper? 
and does it affect result? 
`
class SpatialAttn(nn.Module):
    """Spatial Attention Layer"""
    def __init__(self):
        super(SpatialAttn, self).__init__()

    def forward(self, x):
        # global cross-channel averaging # e.g. 32,2048,24,8
        x = x.mean(1, keepdim=True)  # e.g. 32,1,24,8
        h = x.size(2)
        w = x.size(3)
        x = x.view(x.size(0),-1)     # e.g. 32,192
        z = x
        for b in range(x.size(0)):
            z[b] /= torch.sum(z[b])
        z = z.view(x.size(0),1,h,w)
        return z
`

**Comment by sankin1770** [comment]
I am also quite confused, looking forward to someone to answer。

**Comment by qianyizhang** [comment]
+1
it seems the implementation uses mean while the paper says softmax...

**Comment by TylerYep** [comment]
+1
I made a tensorflow implementation of the original if anyone is interested, but yes, very confused about the softmax vs mean as well.

`class SpatialAttn(Layer):
    ''' Spatial Attention Layer '''
    def __init__(self):
        super().__init__()

    def call(self, x):
        # global cross-channel averaging, e.g. (32,2048,24,8)
        # NOTE: batch_size = None
        batch_size, _, h, w = x.shape
        x = K.mean(x, axis=1, keepdims=True)  # e.g. (32,1,24,8)
        x = Reshape((-1,))(x) # e.g. (32,192)
        row_sum = K.sum(x, axis=0, keepdims=True)
        normalize = K.repeat_elements(row_sum, rep=const.BATCH_SIZE, axis=0)
        z = Lambda(lambda k: k[0] / k[1])([x, normalize])
        z = Reshape((1, h, w))(z)
        return z`

**Comment by allenfutaki** [comment]
The model is trained on such SA module, so the weight is incorrect?
Have anyone tried to use the weight author provided with to evaluate on cuhk03 or market1501?
